### ISSUE TITLE
parserOpts not always passed to conventional-commits-parser

### ISSUE THREAD

**Issue by thet** [release-it/conventional-changelog#37]
My config for the ``conventionalcommits`` preset includes a commit type "breaking" which will then write a "Breaking Changes".
I was wondering why the recommended bump wasn't a major version with commits with that type.
Then I found out that a major level release is recommended when the commit contains a note (See: https://github.com/conventional-changelog/conventional-changelog/blob/840d139c40a2c4e773794d794c34bc4dbfde27c2/packages/conventional-changelog-conventionalcommits/conventional-recommended-bump.js#L20 ). A note is detected via the ``noteKeywords`` config option for ``conventional-commits-parser`` (See: https://github.com/conventional-changelog/conventional-changelog/blob/840d139c40a2c4e773794d794c34bc4dbfde27c2/packages/conventional-commits-parser/lib/regex.js#L21 ).
The parser can be configured, as @release-it/conventional-changelogs points out: https://github.com/release-it/conventional-changelog#parseropts

But it didn't work. I added some console messages to ``getNotesRegex`` in ``conventional-commits-parser/lib/regex.js``:
```
  console.trace();
  console.log(noteKeywords);
```
That gave me the following output:
```
thet@then:/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns$ release-it patch --dry-run
$ npm ping
$ npm whoami
$ npm show @patternslib/patternslib@latest version
$ npm access ls-collaborators @patternslib/patternslib
WARNING Latest version in registry (5.5.1) does not match package.json (5.5.0).
$ git rev-parse --abbrev-ref HEAD
$ git diff --quiet HEAD
$ git rev-parse --abbrev-ref HEAD  [cached]
$ git config --get branch.master.remote
$ git remote get-url origin
! git fetch
$ git describe --tags --match=* --abbrev=0
$ git symbolic-ref HEAD
$ git for-each-ref --format="%(upstream:short)" refs/heads/master
$ git rev-parse --abbrev-ref HEAD  [cached]
$ git config --get branch.master.remote  [cached]
$ git remote get-url origin  [cached]
! git fetch
$ git describe --tags --match=* --abbrev=0  [cached]
WARNING Environment variable "GITHUB_TOKEN" is required for automated GitHub Releases.
WARNING Falling back to web-based GitHub Release.
Trace
    at getNotesRegex (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-commits-parser/lib/regex.js:22:11)
    at module.exports (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-commits-parser/lib/regex.js:54:19)
    at conventionalCommitsParser (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-commits-parser/index.js:74:15)
    at /home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-recommended-bump/index.js:78:15
    at /home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/git-semver-tags/index.js:66:5
    at ChildProcess.exithandler (child_process.js:374:7)
    at ChildProcess.emit (events.js:400:28)
    at maybeClose (internal/child_process.js:1058:16)
    at Socket.<anonymous> (internal/child_process.js:443:11)
    at Socket.emit (events.js:400:28)
[ 'BREAKING CHANGE' ]
/^[\s|*]*(BREAKING CHANGE)[:\s]+(.*)/i
WARNING The recommended bump is "patch", but is overridden with "patch".
Trace
    at getNotesRegex (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-commits-parser/lib/regex.js:22:11)
    at module.exports (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-commits-parser/lib/regex.js:54:19)
    at conventionalCommitsParser (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-commits-parser/index.js:74:15)
    at /home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-changelog-core/index.js:93:15
    at _fulfilled (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/q/q.js:854:54)
    at /home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/q/q.js:883:30
    at Promise.promise.promiseDispatch (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/q/q.js:816:13)
    at /home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/q/q.js:624:44
    at runSingle (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/q/q.js:137:13)
    at flush (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/q/q.js:125:13)
[ 'breaking' ]
/^[\s|*]*(breaking)[:\s]+(.*)/i
$ git log --pretty=format:"* %s (%h)" 5.5.0...HEAD
Trace
    at getNotesRegex (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-commits-parser/lib/regex.js:22:11)
    at module.exports (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-commits-parser/lib/regex.js:54:19)
    at conventionalCommitsParser (/home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-commits-parser/index.js:74:15)
    at /home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/conventional-recommended-bump/index.js:78:15
    at /home/_thet/data/dev/syslab/REPOS/patternslib/Patterns/node_modules/git-semver-tags/index.js:66:5
    at ChildProcess.exithandler (child_process.js:374:7)
    at ChildProcess.emit (events.js:400:28)
    at maybeClose (internal/child_process.js:1058:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:293:5)
[ 'BREAKING CHANGE' ]
/^[\s|*]*(BREAKING CHANGE)[:\s]+(.*)/i

🚀 Let's release @patternslib/patternslib (5.5.0...5.5.1)

[...]
```
Apparently the config is only applied once in three times.

This is my ``.release-it.js`` config:
```
const fs = require("fs");
const path = require("path");

const commits_template = fs
    .readFileSync(path.resolve(".release-it", "conventional-changelog-commit.hbs"))
    .toString();

module.exports = {
    npm: {
        publish: true,
    },
    git: {
        requireBranch: "master",
        commitMessage: "Release new version.",
        commitArgs: ["-n"],
    },
    github: {
        release: true,
    },
    plugins: {
        "@release-it/conventional-changelog": {
            infile: "CHANGES.md",
            preset: {
                name: "conventionalcommits",
                types: [
                    {
                        type: "breaking",
                        section: "Breaking Changes",
                    },
                    {
                        type: "feat",
                        section: "Features",
                    },
                    {
                        type: "fix",
                        section: "Bug Fixes",
                    },
                    {
                        type: "maint",
                        section: "Maintenance",
                    },
                ],
            },
            parserOpts: {
                noteKeywords: ["breaking"],
            },
            writerOpts: {
                commitPartial: commits_template,
            },
        },
    },
};
```

**Comment by thet** [comment]
I also found out that my original problem described in https://github.com/release-it/release-it/issues/833 could be solved by using the config option ``ignoreRecommendedBump`` as described here: https://github.com/release-it/conventional-changelog#ignorerecommendedbump

**Comment by webpro** [comment]
I'm a bit surprised it gets there three times and that the second time the config is being applied.

Looking at this plugin's code, it makes sense it's applied only once: https://github.com/release-it/conventional-changelog/blob/20bf5dccdc1d3f0f4e64c0c6750b4598984d7780/index.js?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L73-L76

Any chance you could do a run without those 4 lines of code, and see whether the options are applied every time? I'm happy to remove them if `conventional-changelog` ignores them on the first `options` argument.

**Comment by webpro** [comment]
You may also have been bitten by #38 (which was just fixed in the latest release-it)

**Comment by webpro** [comment]
Closing this as I think it's fixed. Feel free to open another ticket, though.
